### ISSUE TITLE
Re-enable user on LDAP sync

### ISSUE THREAD

**Issue by hartois** [glpi-project/glpi#993]
If user has not been found in LDAP on sync, one of the actions be performed: keeped/disabled/deleted/remove rights

I use "Disable" action.

But after user has been re-added - nothing done, the user is still be "disabled".
I think that "re-enable user" behavior must be an option.

**Comment by Moiwouam** [comment]
I confirm this issue and there is [a thread in GLPI forum](http://forum.glpi-project.org/viewtopic.php?id=158812) (in french) about it.

**Comment by Eon9D** [comment]
I agree, this would be helpful. objectGUID can make sure it is the same object when they "reappear".
Can this be implemented ?

**Comment by Eon9D** [comment]
@trasher will this be assigned or is it _not planned_

This seems to temporarily work for me (9.2.2).
[ldap.txt](https://github.com/glpi-project/glpi/files/1989509/ldap.txt)

**Comment by minosso** [comment]
> 
> 
> Hi @orthagh
> 
> I am trying your method and now I understand why it was not working for me in September 2017: my Authorizations assignment rule was adding permission but was not re-enabling user. So now it is supposed to work but I have a new issue (probably since 9.2 to 9.3 migration):
> 
> GLPI is not synchronizing anymore the users that are disabled in AD. Note that I always use a ldap filter who excludes them: (&(objectClass=user)(objectCategory=person)(**!(userAccountControl:1.2.840.113556.1.4.803:=2**)))
> 
> But now if I choose a user and do a force synchronization, the last synchronization time is not updated and the authorization are not removed anymore neither.
> Note that the synchronization is still working for active AD users.
> 
> Can you replicate this issue in 9.3.1 ?

I have the same issue. 
New version GLPI 9.2.4

Is there a workaround to fix it and enable this feature again?

**Comment by MFxMF** [comment]
Yes.
Create the rule for Authorizations assignment rules
setup your critria and in the action add action  active==yes

When you will do that, then ldap sync will re-enable user in GLPI, only if when user is not disabled in AD

**Comment by MFxMF** [comment]
> > Hi @orthagh
> > I am trying your method and now I understand why it was not working for me in September 2017: my Authorizations assignment rule was adding permission but was not re-enabling user. So now it is supposed to work but I have a new issue (probably since 9.2 to 9.3 migration):
> > GLPI is not synchronizing anymore the users that are disabled in AD. Note that I always use a ldap filter who excludes them: (&(objectClass=user)(objectCategory=person)(**!(userAccountControl:1.2.840.113556.1.4.803:=2**)))
> > But now if I choose a user and do a force synchronization, the last synchronization time is not updated and the authorization are not removed anymore neither.
> > Note that the synchronization is still working for active AD users.
> > Can you replicate this issue in 9.3.1 ?
> 
> I have the same issue.
> New version GLPI 9.2.4
> 
> Is there a workaround to fix it and enable this feature again?

Yes.
Create the rule for Authorizations assignment rules
setup your critria and in the action add action  active==yes

When you will do that, then ldap sync will re-enable user in GLPI, only if when user is not disabled in AD
